src/lightgbm/src/main/python/LightGBMClassifier.py

```diff
@@ -26,13 +27,40 @@ def saveNativeModel(self, sparkSession, filename):
         """
         Save the booster as string format to a local or WASB remote location.
         """
-        ctx = SparkContext.getOrCreate()
-        sql_ctx = SQLContext.getOrCreate(ctx)
-        jsession = sql_ctx.sparkSession._jsparkSession
+        jsession = sparkSession._jsparkSession
         self._java_obj.saveNativeModel(jsession, filename)
 
+    @staticmethod
+    def loadNativeModelFromFile(sparkSession, filename, labelColName="label",
+                                featuresColName="features", predictionColName="prediction",
+                                probColName="probability", rawPredictionColName="rawPrediction"):
+        """
+        Load the model from a native LightGBM text file.
+        """
+        jsession = sparkSession._jsparkSession
+        ctx = sparkSession.sparkContext
+        loader = ctx._jvm.com.microsoft.ml.spark.LightGBMClassificationModel
+        java_model = loader.loadNativeModelFromFile(jsession, filename, labelColName,
+                                                    featuresColName, predictionColName,
+                                                    probColName, rawPredictionColName)
+        return JavaParams._from_java(java_model)
+
+    @staticmethod
+    def loadNativeModelFromString(sparkSession, model, labelColName="label",
+                                  featuresColName="features", predictionColName="prediction",
+                                  probColName="probability", rawPredictionColName="rawPrediction"):
+        """
+        Load the model from a native LightGBM model string.
+        """
+        ctx = sparkSession.sparkContext

```

Comment:
likewise there is a python equivalent to SparkSession.getOrCreate. Its something like getActiveSession

Reply:
done!